delivery_carrier_label_batch/models/stock_batch_picking.py

```diff
@@ -100,3 +104,26 @@ def create(self, values):
         """
         values = self._values_with_carrier_options(values)
         return super(StockBatchPicking, self).create(values)
+
+    @contextmanager
+    @api.model
+    def _do_in_new_env(self):
+        if tools.config["test_enable"]:
+            yield self.env
+            return
+
+        with api.Environment.manage():
+            with registry(self.env.cr.dbname).cursor() as new_cr:
+                yield api.Environment(new_cr, self.env.uid, self.env.context)
+
+    def purge_tracking_references(self):
+        """Purge tracking for each picking and destination package"""
+        # We need a new env to ensure changes will be applied
+        with self._do_in_new_env() as new_env:
+            for batch in self.with_env(new_env):
+                for pack in batch.move_line_ids.mapped("result_package_id"):
+                    if pack.parcel_tracking:
+                        pack.write({"parcel_tracking": False})
+                for picking in batch.move_line_ids.mapped("picking_id"):
+                    if picking.carrier_tracking_ref:
+                        picking.write({"carrier_tracking_ref": False})
```

Comment:
```suggestion
                packs = self.env["stock.quant.package"].search([("id", "in", batch.move_line_ids.mapped("result_package_id.ids")), ("parcel_tracking", "!=", False)]):
                if packs:
                    packs.write({"parcel_tracking": False})
                pickings = self.env["stock.picking"].search([("move_line_ids", "in", batch.move_line_ids.ids), ("carrier_tracking_ref", "!=", False)])
                if pickings:
                    pickings.write({"carrier_tracking_ref": False})
```
I feel like this could be improved with a `search` + single `write` to avoid the loops. (although it make more sense on the pickings than on the packs)

Not blocking anyway.

Reply:
Why not, i test it.

Reply:
With a little adaptation, it's working well.